State.lua

```diff
@@ -1393,60 +1392,64 @@ do
         return b == nil or ( a.next < b.next )
     end
 
-
     -- Increase max forecast duration because Assassination is pooling hard this tier.
     local FORECAST_DURATION = 10.01
 
-    forecastResources = function( resource )
-
+    forecastResources = function ( resource )
         if not resource then return end
 
+        -- Initialize or wipe tables
+        events = events or {}
+        remains = remains or {}
         wipe( events )
         wipe( remains )
 
-        local now = state.now + state.offset -- roundDown( state.now + state.offset, 2 )
-
-        local timeout = FORECAST_DURATION * state.haste -- roundDown( FORECAST_DURATION * state.haste, 2 )
+        local now = state.now + state.offset
+        local timeout = FORECAST_DURATION * state.haste
 
         if state.class.file == "DEATHKNIGHT" and state.runes then
-            timeout = max( timeout, 0.01 + 2 * state.runes.cooldown )
+            -- Adjust timeout based on rune cooldowns and regen models for Frost DK
+            timeout = max( timeout, 0.01 + state.runes.cooldown * ( 6 - state.runes.current ) )
         elseif state.spec.assassination then
             timeout = 15.01
         end
 
         timeout = timeout + state.gcd.remains
-
         local r = state[ resource ]
 
-        -- We account for haste here so that we don't compute lots of extraneous future resource gains in Bloodlust/high haste situations.
+        -- Initialize forecast tables
         remains[ resource ] = timeout
+        r.times = r.times or {}
+        r.values = r.values or {}
+        r.forecast = r.forecast or {}
```

Comment:
These are technically sane, but these tables are all created upon `RegisterResource` so this shouldn't be needed.

Reply:
Maybe all my nil table issues were a scope thing, I'll take a look at them all.